src/clusterfuzz/_internal/bot/tasks/setup.py

```diff
@@ -280,21 +284,42 @@ def _get_testcase_file_and_path(testcase):
   return input_directory, testcase_path
 
 
-def unpack_testcase(testcase):
-  """Unpack a testcase and return all files it is composed of."""
-  # Figure out where the testcase file should be stored.
-  input_directory, testcase_file_path = _get_testcase_file_and_path(testcase)
+def get_signed_testcase_download_url(testcase):
+  """Returns a signed download URL for the testcase."""
+  key, _ = _get_testcase_key_and_archive_status(testcase)
+  return blobs.get_signed_download_url(key)
+
 
-  minimized = testcase.minimized_keys and testcase.minimized_keys != 'NA'
-  if minimized:
+def _get_testcase_key_and_archive_status(testcase):
+  """Returns the testcase's key and whether or not it is archived."""
+  if _is_testcase_minimized(testcase):
     key = testcase.minimized_keys
     archived = bool(testcase.archive_state & data_types.ArchiveStatus.MINIMIZED)
-  else:
-    key = testcase.fuzzed_keys
-    archived = bool(testcase.archive_state & data_types.ArchiveStatus.FUZZED)
+    return key, archived
+
+  key = testcase.fuzzed_keys
+  archived = bool(testcase.archive_state & data_types.ArchiveStatus.FUZZED)
+  return key, archived
+
+
+def _is_testcase_minimized(testcase):
+  return testcase.minimized_keys and testcase.minimized_keys != 'NA'
+
+
+def download_testcase(key, testcase_download_url, dst):
```

Comment:
Are we required to write doc-strings for small functions (like this one and the one above)?
I feel they are straightforward enough and do not need any notes, but I am unsure if doc-string is mandatory.

Reply:
I think it's only really mandatory when linting requires it.
otherwise I would document if it's not obvious and/or widely used. This is so short and not-widely used that it's easier to read the code.